src/layers/VivViewerLayer/VivViewerLayer.js

```diff
@@ -108,25 +91,37 @@ export default class VivViewerLayer extends CompositeLayer {
   }
 
   renderLayers() {
-    const {
-      connections,
-      pool,
-      imageWidth,
-      imageHeight,
-      tileSize,
-      minZoom
-    } = this.state;
+    const { imageWidth, imageHeight, tileSize, minZoom } = this.state;
 
+    const { loader } = this.props;
     const layerProps = this._overrideChannelProps();
-    const layers = connections
+    const getTileData = ({ x, y, z }) => {
+      if (
+        isInTileBounds({
+          x,
+          y,
+          z: -z,
+          imageWidth,
+          imageHeight,
+          minZoom,
+          tileSize
+        })
+      ) {
+        return loader.getTile({
+          x,
+          y,
+          z: -z
+        });
+      }
+      return null;
+    };
+    const layers = loader
```

Comment:
Do we need to check if the loader exists anymore since it is being passed in directly?  On that topic, I think we can get rid of the above `updateState` function and just move those props into here.  They don't need to be in state anymore, I don't think.